subsys/bluetooth/host/l2cap.c

```diff
@@ -2002,10 +1953,29 @@ static int l2cap_chan_le_send(struct bt_l2cap_le_chan *ch,
 	/* Save state so it can be restored if we failed to send */
 	net_buf_simple_save(&buf->b, &state);
 
-	seg = l2cap_chan_create_seg(ch, buf, sdu_hdr_len);
-	if (!seg) {
-		atomic_inc(&ch->tx.credits);
-		return -EAGAIN;
+	if (net_buf_frags_len(buf) <= ch->tx.mps) {
+		LOG_DBG("len < MPS, not allocating seg for %p", buf);
```

Comment:
```suggestion
	if (net_buf_frags_len(buf) <= ch->tx.mps) {
		LOG_DBG("len <= MPS, not allocating seg for %p", buf);
```

Small mismatch between code and comment